### PR TITLE
Fix #5578 namespaces/use placement

### LINKED ISSUE #5578: PHP Fix Imports issues with namespace/use placing
### Apache NetBeans version

Apache NetBeans latest daily build

### What happened

There are many issues related to inappropriate place of namespace declarations such as multiple php open tag, declare, not empty first line. Below I'll try to explain the problems and also show the result after possible pull request changes:

1) Import in files with mixed content (php/html) or where is no php code on the first few lines:

-- initial state
Screen Shot 2023-02-28 at 15 21 50
-- after import
Screen Shot 2023-02-28 at 13 33 14
-- after fix
Screen Shot 2023-02-28 at 15 42 26

2) declare(ticks=int) inside class or other block structure:

--initial state
Screen Shot 2023-02-28 at 16 12 28
--after import
Screen Shot 2023-02-28 at 16 12 38
--after fix
Screen Shot 2023-02-28 at 16 13 57

3) Import when declare has block structure or block structure (like function) on first line of the file:

--initial state
Screen Shot 2023-02-28 at 13 35 09
--after import
Screen Shot 2023-02-28 at 13 35 28
--after fix (for strict_types declare namespace should and would be after the declare statement, but without strict_types identifier declare could be everywhere)
Screen Shot 2023-02-28 at 15 49 05

4) When namespace block is one line in file with multiple php tag blocks:

--initial state
Screen Shot 2023-02-28 at 15 38 16
--after import
Screen Shot 2023-02-28 at 15 38 31
--after fix
Screen Shot 2023-02-28 at 15 49 23

5) Multiline comment after namespace or after last use element:

--initial state
Screen Shot 2023-02-28 at 13 43 33
--after import
Screen Shot 2023-02-28 at 13 43 43
--after fix
Screen Shot 2023-02-28 at 15 49 44

6) Not so important, but still an issue related to inappropriate detection of use/namespace elements borders:

--initial state
Screen Shot 2023-02-28 at 15 40 07
--after import
Screen Shot 2023-02-28 at 15 40 49
--after fix
Screen Shot 2023-02-28 at 15 50 17

7) Import with USE in multiple namespaces removes all USES in other namespaces:

--initial state
Screen Shot 2023-03-01 at 23 49 07
--after import
Screen Shot 2023-03-01 at 23 49 19
--after fix
Screen Shot 2023-03-02 at 00 09 22

### How to reproduce

Go and run Source->Fix Imports or use Cmd(Ctrl)+Shift+i keys on any of the examples above.

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

MacOS Mojave Version 10.14.6

### JDK

13.0.2

### Apache NetBeans packaging

Own source build

### Anything else

I used last build for checks, but did a few in Netbeans 14 with same result.

### Are you willing to submit a pull request?

Yes

### PR DESCRIPTION
**Pull request by rossluk**
Fix "Fix Import" namespaces/uses placement in multiple php open tag files, when first line of php file is not empty or it is a block element, after declare statements in different scope

### CODE REVIEW

File: php/php.editor/src/org/netbeans/modules/php/editor/actions/FixUsesPerformer.java
Comment (lbownik) [inline php/php.editor/src/org/netbeans/modules/php/editor/actions/FixUsesPerformer.java:165]: would return in catch block read better?

Comment (rossluk) [reply php/php.editor/src/org/netbeans/modules/php/editor/actions/FixUsesPerformer.java:165]: i am not a java developer, if it would be better i am willing to make changes, but just checked few files with similar logic and there is usually return outside

File: php/php.editor/src/org/netbeans/modules/php/editor/actions/FixUsesPerformer.java
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/actions/FixUsesPerformer.java:159]: Formatting: Please move "{" to the end of the previous line.

File: php/php.editor/src/org/netbeans/modules/php/editor/actions/FixUsesPerformer.java
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/actions/FixUsesPerformer.java:526]: Please add `// NOI18N`

**Comment by junichi11** [comment]
Could you add unit tests?

**Review by tmysik — CHANGES_REQUESTED** [review]
Please, add unit tests for your use case. Thank you.

**Comment by rossluk** [comment]
I've found more issues related to use placement, will do force push when have it fixed

**Comment by rossluk** [comment]
Added new test case and refactored some code. There is still a problem with multiple global namespaces, but problem is not related to Fix Import and I suppose should be fixed by other PR.

**Comment by junichi11** [comment]
Please rename `testNB5578` to `testGH5578`.
Please add issue number to the commit message.

**Review by tmysik — APPROVED** [review]
LGTM, thanks for your work!

**Comment by tmysik** [comment]
@rossluk Please, wait for approval from @junichi11. Thank you.

File: php/php.editor/src/org/netbeans/modules/php/editor/actions/FixUsesPerformer.java
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/actions/FixUsesPerformer.java:553]: Please add 
```java
            if (CancelSupport.getDefault().isCancelled()) {
                return;
            }
```

File: php/php.editor/test/unit/data/testfiles/actions/testGH5578/01/testGH5578_01.php.fixUses
Comment (junichi11) [inline php/php.editor/test/unit/data/testfiles/actions/testGH5578/01/testGH5578_01.php.fixUses:7]: We should add use statements below the declare statement if possible.
see: https://www.php-fig.org/psr/psr-12/#3-declare-statements-namespace-and-import-statements

Please add a test case that has also `namespace` (`declare` and `namespace`).

Comment (rossluk) [reply php/php.editor/test/unit/data/testfiles/actions/testGH5578/01/testGH5578_01.php.fixUses]: Updated logic due to it, added new tests, also added a check for USE placing after phpDoc if there is no namespace in file (if namespace and phpDoc are there it will be placed after namespace).

File: php/php.editor/src/org/netbeans/modules/php/editor/actions/FixUsesPerformer.java
Comment (rossluk) [reply php/php.editor/src/org/netbeans/modules/php/editor/actions/FixUsesPerformer.java:564]: Done

**Review by junichi11 — APPROVED** [review]
Thank you for your work.

**Pull request merged**
